instrumentation/resque/lib/opentelemetry/instrumentation/resque/instrumentation.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Resque
+      # The Instrumentation class contains logic to detect and install the Resque instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        install do |_config|
+          require_dependencies
+          patch
+        end
+
+        present do
+          defined?(::Resque)
+        end
+
+        option :job_class_span_names, default: false, validate: :boolean
+        option :enqueuer_propagation, default: false, validate: :boolean
```

Comment:
Not sure if this config name is too cryptic, open to alternatives.

Reply:
Also need to add this config option to sidekiq.

Reply:
I called it `context_propagation`, but that might be similarly vague / cryptic. I'll be happy to standardize on whatever we decide, though.

Reply:
What's a succinct way of saying "this config options continues the trace from the enqueuing span to the job process span"?  😄

Reply:
```rb
        option :continue_trace_from_enqueuing_span_to_job_process_span, default: false, validate: :boolean
```

Seems a little wordy

Reply:
Yeah .... that's quite the option name there. Perhaps `propagation_style` ?

Reply:
Apparently config option naming is where I like to bikeshed 😄 

* `enqueuer_propagation` is so so
* `context_propagation` is better
* `propagation_style` leads me to believe I would need to pass in something more than a boolean
* `propagate_context` ... this one I like :)

Reply:
`propagation_style` <- to be fair, in #726 we accept `:link`, `:child`, or `:none` 😆

Reply:
I like `propagation_style` with the symbol options - much clearer than the boolean flag.

Reply:
Updated to use `propagation_style` and to match the behaviour from #726